diff --git a/vendor/magento/module-configurable-product/view/frontend/web/js/options-updater.js b/vendor/magento/module-configurable-product/view/frontend/web/js/options-updater.js
index 558a1fdf310..6f187983031 100644
--- a/vendor/magento/module-configurable-product/view/frontend/web/js/options-updater.js
+++ b/vendor/magento/module-configurable-product/view/frontend/web/js/options-updater.js
@@ -61,6 +61,7 @@ define([
             this.setProductOptions(cartData());
             this.updateOptions();
         }.bind(this));
+        this.updateOptions();
     },
 
     /**
diff --git a/vendor/magento/module-msrp/view/base/web/js/msrp.js b/vendor/magento/module-msrp/view/base/web/js/msrp.js
index deeadd9b55b..72dd1d8bbec 100644
--- a/vendor/magento/module-msrp/view/base/web/js/msrp.js
+++ b/vendor/magento/module-msrp/view/base/web/js/msrp.js
@@ -73,6 +73,7 @@ define([
                 this.initTierPopup();
             }
             $(this.options.cartButtonId).on('click', this._addToCartSubmit.bind(this));
+            $(this.options.cartForm).on('submit', this._onSubmitForm.bind(this));
         },
 
         /**
@@ -249,8 +250,10 @@ define([
 
         /**
          * Handler for addToCart action
+         *
+         * @param {Object} e
          */
-        _addToCartSubmit: function () {
+        _addToCartSubmit: function (e) {
             this.element.trigger('addToCart', this.element);
 
             if (this.element.data('stop-processing')) {
@@ -266,8 +269,20 @@ define([
             if (this.options.addToCartUrl) {
                 $('.mage-dropdown-dialog > .ui-dialog-content').dropdownDialog('close');
             }
+
+            e.preventDefault();
             $(this.options.cartForm).submit();
+        },
 
+        /**
+         * Handler for submit form
+         *
+         * @private
+         */
+        _onSubmitForm: function () {
+            if ($(this.options.cartForm).valid()) {
+                $(this.options.cartButtonId).prop('disabled', true);
+            }
         }
     });
 
diff --git a/vendor/magento/module-product-video/view/frontend/web/js/fotorama-add-video-events.js b/vendor/magento/module-product-video/view/frontend/web/js/fotorama-add-video-events.js
index edf02c19710..b46aee57361 100644
--- a/vendor/magento/module-product-video/view/frontend/web/js/fotorama-add-video-events.js
+++ b/vendor/magento/module-product-video/view/frontend/web/js/fotorama-add-video-events.js
@@ -179,12 +179,14 @@ define([
          * @private
          */
         clearEvents: function () {
-            this.fotoramaItem.off(
-                'fotorama:show.' + this.PV +
-                ' fotorama:showend.' + this.PV +
-                ' fotorama:fullscreenenter.' + this.PV +
-                ' fotorama:fullscreenexit.' + this.PV
-            );
+            if (this.fotoramaItem !== undefined) {
+                this.fotoramaItem.off(
+                    'fotorama:show.' + this.PV +
+                    ' fotorama:showend.' + this.PV +
+                    ' fotorama:fullscreenenter.' + this.PV +
+                    ' fotorama:fullscreenexit.' + this.PV
+                );
+            }
         },
 
         /**
diff --git a/vendor/magento/module-swatches/view/frontend/web/js/swatch-renderer.js b/vendor/magento/module-swatches/view/frontend/web/js/swatch-renderer.js
index 50a3af34aa6..7289c50894f 100644
--- a/vendor/magento/module-swatches/view/frontend/web/js/swatch-renderer.js
+++ b/vendor/magento/module-swatches/view/frontend/web/js/swatch-renderer.js
@@ -1184,7 +1184,10 @@ define([
                 }
 
                 imagesToUpdate = this._setImageIndex(imagesToUpdate);
-                gallery.updateData(imagesToUpdate);
+
+                if (!_.isUndefined(gallery)) {
+                    gallery.updateData(imagesToUpdate);
+                }
 
                 if (isInitial) {
                     $(this.options.mediaGallerySelector).AddFotoramaVideoEvents();
